apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -35,7 +36,25 @@ public sealed class SpannerConnectionStringBuilder : DbConnectionStringBuilder
         public string DataSource
         {
             get => GetValueOrDefault("Data Source");
-            private set => this["Data Source"] = value;
+            private set => this["Data Source"] = ValidatedDataSource(value);
+        }
+
+        private string ValidatedDataSource(string dataSource)
+        {
+            var parts = dataSource?.Split('/');
+            if (parts?.Length == 3)
+            {
+                return dataSource;
+            }
+            if (parts?.Length == 6
+                && String.Equals(parts[0], "projects", StringComparison.OrdinalIgnoreCase)
+                && String.Equals(parts[2], "instances", StringComparison.OrdinalIgnoreCase)
+                && String.Equals(parts[4], "databases", StringComparison.OrdinalIgnoreCase))
+            {
+                return dataSource;
+            }
+            throw new ArgumentException($"'{dataSource}' is not a valid value for ${nameof(DataSource)}.  It should be of the form "
+                + "projects/<project>/instances/<instance>/databases/<database>.", nameof(DataSource));
```

Comment:
Can something like [PathTemplate](https://github.com/googleapis/gax-dotnet/blob/master/src/Google.Api.Gax/PathTemplate.cs#L29) can be used here?